frontend/src/pages/AddProduct/index.js

```diff
@@ -12,105 +14,135 @@ const initialProductState = {
   categoryId: "",
 }
 
-const AddProduct = () => {
-  const [product, setProduct] = useState(initialProductState)
-
-  const handleInputChange = (event) => {
-    const { name, value } = event.target
-    setProduct({ ...product, [name]: value })
-  }
-
-  const createProduct = (event) => {
-    event.preventDefault()
-
-    ProductDataService.create(product).then(() => {
+const schema = yup.object().shape({
+  title: yup.string().required(),
+  description: yup.string().required(),
+  price: yup.number().required().positive(),
+  stock: yup.number().required().positive().integer(),
+  isActive: yup.boolean().required(),
+  categoryId: yup.number().required().positive().integer(),
+})
+const createProduct = async (values, { resetForm }) => {
+  return ProductDataService.create(values)
+    .then(() => {
       toast.success("Produto cadastrado com sucesso!!")
       resetForm()
     })
-  }
-
-  const resetForm = () => {
-    setProduct(initialProductState)
-  }
-
+    .catch((err) => {
+      toast.error("Erro ao cadastrar produto")
+    })
+}
+const AddProduct = () => {
   return (
     <Container>
-      <FormContainer onSubmit={createProduct} onReset={resetForm}>
-        <div>
-          <h2>Cadastrar Produtos</h2>
-          <div className="line"></div>
-          <div>
-            <div>
-              <label>Nome</label> <br />
-              <input
-                type="text"
-                id="title"
-                required
-                value={product.title}
-                onChange={handleInputChange}
-                name="title"
-                placeholder="Coqueteleira"
-              />
-            </div>
+      <Formik
+        initialValues={initialProductState}
+        validationSchema={schema}
+        onSubmit={createProduct}
+      >
+        {({ errors, touched }) => (
+          <FormContainer>
             <div>
-              <label>Descrição</label> <br />
-              <input
-                type="text"
-                id="description"
-                required
-                value={product.description}
-                onChange={handleInputChange}
-                name="description"
-                placeholder="Ex: Coqueteleira 500ml de aço inox"
-              />
+              <h2>Cadastrar Produtos</h2>
+              <div className="line"></div>
+              <div>
+                <div>
+                  <label>Nome</label> <br />
+                  <Field
+                    type="text"
+                    id="title"
+                    required
+                    name="title"
+                    placeholder="Coqueteleira"
+                    className={errors.title && touched.title ? "has-error" : ""}
+                  />
+                  {errors.title && touched.title ? (
+                    <span className="error">{errors.title}</span>
+                  ) : (
+                    ""
+                  )}
```

Comment:
Legal que tem usado bastante os condicionais ternários, mas acredito que o código poderia ficar mais simplificado se você usasse o 'short-circuit'. Como ficaria em um exemplo pequeno:

```suggestion
                    className={errors.title && touched.title && "has-error"}
                  />
                  {errors.title && touched.title && (
                    <span className="error">{errors.title}</span>
                  )}
```

Um artigo explicando como isso funciona: https://codeburst.io/javascript-what-is-short-circuit-evaluation-ff22b2f5608c

Reply:
Atualizado de acordo com a sugestão em todos os inputs
![image](https://user-images.githubusercontent.com/45775182/144287794-fbd83172-77ca-4fa7-8353-e7c9e5c7ec55.png)